### PR TITLE
graph: user groupTypes property to indicate read-only groups

### PR DESCRIPTION
**Pull request by rhafer**
By setting GRAPH_LDAP_GROUP_CREATE_BASE_DN a distinct subtree can be configure
where new LDAP groups are created. That subtree needs to be subordinate to
GRAPH_LDAP_GROUP_BASE_DN. All groups outside for
GRAPH_LDAP_GROUP_CREATE_BASE_DN are considered read-only and only group below
that DN can be updated and deleted.
    
This is introduced for a pretty specific usecase where most groups are managed
in an external source (e.g. a read-only replica of an LDAP tree). But we still
want to allow the local administrator to create groups in a writeable subtree
attached to that replica.

To test this you can start ocis with

```
IDM_CREATE_DEMO_USERS=true GRAPH_LDAP_GROUP_CREATE_BASE_DN=ou=users,o=libregraph-idm GRAPH_LDAP_GROUP_BASE_DN=o=libregraph-idm ocis/bin/ocis server
```

This will make all the demo groups "readonly" (indicated through "ReadOnly" value in "groupTypes":

```
~> curl -s -k -u admin:admin  'https://localhost:9200/graph/v1.0/groups/users'   -H 'accept: application/json' | jq .
{
  "displayName": "users",
  "groupTypes": [
    "ReadOnly"
  ],
  "id": "509a9dcd-bb37-4f4f-a01a-19dca27d9cfa"
}
```

Trying to update the group will end with an error (currently a 500), but I still need to fix that to return a 405 or 403.

Adding groups is still possible, but they'll endup in a seperate part of the LDAP tree:

```
~> curl -s -k -u admin:admin  'https://localhost:9200/graph/v1.0/groups'   -H 'accept: application/json' -X POST --data '{"displayName": "Lokal"}' | jq .
{
  "displayName": "Lokal",
  "groupTypes": [],
  "id": "0a0f201c-c38f-40fd-a4c4-cd87d9168ee9"
}
```

Groups without the "ReadOnly" value in the groupTypes can be updated:

```
~> curl -i -k -u admin:admin  'https://localhost:9200/graph/v1.0/groups/0a0f201c-c38f-40fd-a4c4-cd87d9168ee9'   -H 'accept: application/json' -X PATCH --data '{"displayName": "Lokale Gruppe"}' 
HTTP/1.1 204 No Content
Date: Thu, 30 Mar 2023 16:48:47 GMT
X-Graph-Version: 3.0.0-alpha.1+093d65dcd
```

### CODE REVIEW

**Comment by rhafer** [comment]
@JammingBen This should be of interest for you.

Note: We opted for using the `groupTypes` list property for indicating the readOnly state of a group as we didn't want to introduce some incompatible (with MS graph) property on the groups resource. If that is too clumsy for you I am open for discussion. But I guess @micbar and @butonic have some opinion about this as well.

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/20982/68/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20982/tracing/file-alice-2023-3-30-05-05-40.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20982/tracing/space-members-can-be-invited-with-an-expiration-date-alice-2023-3-30-05-07-15.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20982/tracing/space-members-can-be-invited-with-an-expiration-date-brian-2023-3-30-05-07-21.zip 


</p></details>

**Comment by JammingBen** [comment]
> @JammingBen This should be of interest for you.
> 
> Note: We opted for using the `groupTypes` list property for indicating the readOnly state of a group as we didn't want to introduce some incompatible (with MS graph) property on the groups resource. If that is too clumsy for you I am open for discussion. But I guess @micbar and @butonic have some opinion about this as well.

That's okay, I can work with that. Thanks for the heads-up!

File: changelog/unreleased/enhancement-ldap-group-create-basedn.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Enhancement: Make the LDAP base DN for new groups configurable
+
+The LDAP backend for the Graph service introduced a new config option for setting the
+Parent DN for new groups created via the `/groups/` endpoint. (`GRAPH_LDAP_GROUP_CREATE_BASE_DN`)
+
+It defaults to the value of `GRAPH_LDAP_GROUP_BASE_DN`. If the set to a different value the
```
Comment (kobergj) [inline changelog/unreleased/enhancement-ldap-group-create-basedn.md:6]:
```suggestion
It defaults to the value of `GRAPH_LDAP_GROUP_BASE_DN`. If set to a different value the
```

**Review by kobergj — APPROVED** [review]

**Pull request merged**